compiler/src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -538,7 +538,8 @@ class Namer { typer: Typer =>
             trees.map {
               case mcls @ TypeDef(name, impl: Template) if valid(mcls) =>
                 moduleClsDef.get(name) match {
-                  case Some((stat1, mcls1@TypeDef(_, impl1: Template))) =>
+                  case Some((stat1, mcls1)) =>
+                    assert(mcls1.rhs.isInstanceOf[Template])
```

Comment:
It will still throw an exception, the absent warning is misleading.
What did you intend to achive?
If you wanted to remove warning, you should use `(moduleClsDef.get(name): @unchecked)` instead